test/integration/services/data/test-get-capacity-breakdown.js

```diff
@@ -0,0 +1,98 @@
+
+const dataHelper = require('../../../helpers/data/aggregated-data-helper')
+const getCapacityBreakdown = require('../../../../app/services/data/get-capacity-breakdown')
+
+var inserts = []
+
+describe('services/data/get-capacity-breakdown', function () {
+  before(function () {
+    return dataHelper.addWorkloadCapacitiesForOffenderManager()
+      .then(function (builtInserts) {
+        inserts = builtInserts
+      })
+  })
+
+  it('should retrieve all workload_owners workloads within a team', function () {
+    return getCapacityBreakdown(inserts.filter((item) => item.table === 'team')[0].id, 'team')
+      .then(function (results) {
+        var expectedEntry = {
+          total_points: 50,
+          available_points: 25,
+          reduction_hours: 3,
+          linkId: inserts.filter((item) => item.table === 'workload_owner')[0].id,
+          name: 'Test_Forename Test_Surname',
+          grade: 'PO',
+          totalCases: 5,
+          cmsReductionHours: 0,
+          contractedHours: 37.5
+        }
+
+        expect(results.length).to.be.eql(2)
+        expect(results).to.contain(expectedEntry)
+      })
+  })
+
+  it('should retrieve summed capacity breakdown data for teams within ldu', function () {
+    return getCapacityBreakdown(inserts.filter((item) => item.table === 'ldu')[0].id, 'ldu')
+    .then(function (results) {
+      var expectedEntry = {
+        total_points: 70,
+        available_points: 35,
+        reduction_hours: 6,
+        linkId: inserts.filter((item) => item.table === 'team')[0].id,
+        name: 'Test Team',
+        grade: 'PO',
+        totalCases: 10,
+        cmsReductionHours: 0,
+        contractedHours: 75
+      }
+
+      expect(results.length).to.be.eql(1)
+      expect(results).to.contain(expectedEntry)
+    })
+  })
+
+  it('should retrieve summed capacity breakdown data for ldus within region', function () {
+    return getCapacityBreakdown(inserts.filter((item) => item.table === 'region')[0].id, 'region')
+    .then(function (results) {
+      var expectedEntry = {
+        total_points: 70,
+        available_points: 35,
+        reduction_hours: 6,
+        linkId: inserts.filter((item) => item.table === 'ldu')[0].id,
+        name: 'Test LDU',
+        grade: 'PO',
+        totalCases: 10,
+        cmsReductionHours: 0,
+        contractedHours: 75
+      }
+
+      expect(results.length).to.be.eql(1)
+      expect(results).to.contain(expectedEntry)
+    })
+  })
+
+  it('should retrieve summed capacity breakdown data for regions within national', function () {
+    return getCapacityBreakdown(undefined, 'hmpps')
+    .then(function (results) {
+      var expectedEntry = {
+        total_points: 70,
+        available_points: 35,
+        reduction_hours: 6,
+        linkId: inserts.filter((item) => item.table === 'region')[0].id,
+        name: 'Test Region',
+        grade: 'PO',
+        totalCases: 10,
+        cmsReductionHours: 0,
+        contractedHours: 75
```

Comment:
Three of these expected entries are the same except for `linkId`? Define in one place and use Object.assign for the `linkId`?

Reply:
changed